src/app/dynamic/enterprise/quotas/component.ts

```diff
@@ -0,0 +1,142 @@
+//    and distribution, is expressly prohibited.
+// 3. THE SOFTWARE IS PROVIDED “AS IS”, WITHOUT WARRANTY OF ANY KIND,
+//    EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+//    MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+//    IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY
+//    CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+//    TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+//    SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+//
+// END OF TERMS AND CONDITIONS
+
+import {Component, OnChanges, OnInit, ViewChild} from '@angular/core';
+import {MatTableDataSource} from '@angular/material/table';
+import {MatPaginator} from '@angular/material/paginator';
+import {MatSort, Sort} from '@angular/material/sort';
+
+import {take, takeUntil, tap} from 'rxjs/operators';
+import {Subject} from 'rxjs';
+
+import {SettingsService} from '@core/services/settings';
+import {UserService} from '@core/services/user';
+
+import {getNumberFromString} from '@shared/utils/common';
+import {Member} from '@shared/entity/member';
+import {Quota} from '@shared/entity/quota';
+
+import * as _ from 'lodash';
+
+enum Column {
+  projectId = 'projectId',
+  CPU = 'CPU',
+  memory = 'memory',
+  storage = 'storage',
+  spacer = 'spacer',
+}
+
+@Component({
+  selector: 'km-quotas',
+  templateUrl: './template.html',
+  styleUrls: ['style.scss'],
+})
+export class QuotasComponent implements OnInit, OnChanges {
+  user: Member;
+  quotas: Quota[] = [];
+  dataSource = new MatTableDataSource<Quota>(this.quotas);
+  displayedColumns: string[] = [Column.projectId, Column.CPU, Column.memory, Column.storage, Column.spacer];
+
+  isLoading = true;
+  Column = Column;
+
+  @ViewChild(MatSort, {static: true}) sort: MatSort;
+  @ViewChild(MatPaginator, {static: true}) paginator: MatPaginator;
+
+  private _unsubscribe = new Subject<void>();
+
+  constructor(private readonly _settingsService: SettingsService, private readonly _userService: UserService) {}
+
+  ngOnInit(): void {
+    this.dataSource.sort = this.sort;
+    this.sort.active = Column.projectId;
+    this.sort.direction = 'asc';
+
+    this._settingsService.quotas
+      .pipe(
+        tap(() => (this.isLoading = false)),
+        takeUntil(this._unsubscribe)
+      )
+      .subscribe({
+        next: quotas => {
+          if (_.isEqual(quotas, this.quotas)) {
+            return;
+          }
+
+          this.quotas = quotas;
+          this.setDataSource();
+        },
+      });
+
+    this.dataSource.paginator = this.paginator;
+
+    this._userService.currentUserSettings.pipe(takeUntil(this._unsubscribe)).subscribe(settings => {
+      this.paginator.pageSize = settings.itemsPerPage;
+      this.dataSource.paginator = this.paginator; // Force refresh.
+    });
+
+    this._userService.currentUser.pipe(take(1)).subscribe(user => (this.user = user));
+  }
+
+  ngOnChanges(): void {
```

Comment:
This component doesn't have any `@Input` so why do need `ngOnChanges` method?

Reply:
This logic is copied over from `admin/accounts` and  `admin/admins`, and it is present in many other admin setting components too

Reply:
changes implemented